### ISSUE TITLE
api_surface not working with CA1051

### ISSUE THREAD

**Issue by giggio** [dotnet/roslyn-analyzers#3590]
#### Analyzer package

[Microsoft.CodeAnalysis.FxCopAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.FxCopAnalyzers)

#### Package Version

v3.0.0 (Latest)

#### Diagnostic ID

Example: [CA1051](https://docs.microsoft.com/en-us/visualstudio/code-quality/ca1051)

#### Repro steps

1. Declare a protected field
2. Add `dotnet_code_quality.CA1051.api_surface = private` to `.editorconfig` (2 folders above)
3. Compile and see the warning

#### Expected behavior

No diagnostics reported.

#### Actual behavior

Diagnostic reported as:

````
Error	CA1051	Do not declare visible instance fields	Proj	C:\p\file.cs	13	Active	The primary use of a field should be as an implementation detail. Fields should be private or internal and should be exposed by using properties.
````

**Comment by mavasani** [comment]
Can you try latest 3.3.x beta from https://dotnet.myget.org/feed/roslyn-analyzers/package/nuget/Microsoft.CodeAnalysis.FxCopAnalyzers? Prior to 3.3, we only supported configuration as per https://github.com/dotnet/roslyn-analyzers/blob/master/docs/Analyzer%20Configuration.md#prior-to-vs2019-163

**Comment by mavasani** [comment]
https://github.com/dotnet/roslyn-analyzers/pull/3591/files to clarify the documentation

**Comment by giggio** [comment]
Yes, that fixes it. I'll wait for the stable release of 3.3. Would you know when it is exepected, @mavasani? Will package version jump from 3.0 to 3.3, or is it really far along?
I see the updated docs, that also clarifies it. I was on latest VS, but on a not fixed package version.
Thanks!

**Comment by mavasani** [comment]
The analyzer package version is basically indicating the minimum compiler API version that the analyzer depends on. EditorConfig support was first added to Roslyn compiler in compiler version 3.3 (VS version VS2019 16.3), and supporting this in analyzer package needs us move to compiler 3.3, hence the jump from 3.0.0 to 3.3.0 in analyzer package version.

The stable 3.3.x release is not yet planned, but hopefully we will soon do a pre-release.
